backend/main.py

```diff
@@ -72,30 +69,12 @@ def preprocessing():
     Preprocess data for all strategy and save results locally,
     each strategy in a file.
     """
-    for strategy in STRATEGIES:
-        output_filename = utils.generate_filename_on_strategy(
-            FILENAME, strategy)
-        if os.path.isfile(output_filename):
-            continue
-        data = downsample.downsample(
-            FILENAME, strategy, NUMBER_OF_RECORDS_PER_SECOND)
-        data_csv = utils.convert_to_csv(data)
-        if data_csv is None:
-            return 'Empty records', 500
-        with open(output_filename, 'w') as filewriter:
-            filewriter.write(data_csv)
-            filewriter.flush()
-    return 'Preprocessing Successful!'
-
-
-@app.before_first_request
-def initialize():
-    """Initializes with preprocessing.
-
-    Preprocess all power data at the first request.
-    """
-    # TODO(tangyifei@): Initialize at the start of service, instead of first request.
-    preprocessing()
+    name = request.args.get('name', type=str)
+    rate = request.args.get('name', type=int)
+    success = downsample.preprocess(name, rate)
+    if success:
+        return 'Preprocessing Successful!', 200
+    return 'Preprocessing Incomplete', 400
```

Comment:
Shall we give failure reasons here? e.g. the file with given name doesn not exist; the file is not in csv format, etc

Reply:
same reason ^